### ISSUE TITLE
Error while using with angular 9 

### ISSUE THREAD

**Issue by ipranjal** [mqttjs/MQTT.js#1105]
``` 
Uncaught TypeError: Cannot read property 'slice' of undefined
    at Object../node_modules/mqtt/node_modules/readable-stream/lib/_stream_writable.js (_stream_writable.js:57)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/mqtt/node_modules/readable-stream/readable-browser.js (readable-browser.js:4)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/mqtt/lib/store.js (store.js:8)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/mqtt/lib/client.js (client.js:7)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/mqtt/lib/connect/index.js (index.js:3)
    at __webpack_require__ (bootstrap:84)
```

**Comment by slee981** [comment]
In `polyfills.ts` try adding: 

```javascript
...

global.process = require('process');
```

**Comment by haroldirvingpumapolo** [comment]
to all node module files ending in js, that have this line of code (search for it with the global visual code finder)

var asyncWrite = !process.browser && ['v0.10', 'v0.9.'].indexOf(process.version.slice(0, 5)) > -1? setImmediate: pna.nextTick;

and replaced with this

var asyncWrite = !process.browser && ['v0.10', 'v0.9.'].indexOf(process.version?.slice(0, 5)) > -1? setImmediate: pna.nextTick;

In conclusion just add a question mark.

This error is very strange because the application worked for all my coworkers except me, so to continue working I changed this (I know it is not the best solution).
